# Práctica Clase 2

Utilizando Python cálcular las siguientes probabilidades. Para cada uno intenta determinar que método aplicaste.<br>


In [4]:
import math
import numpy as np
import pandas as pd

1. Lanza una moneda al aire 10 veces, ¿cuantos resultados posibles forman parte del espacio muestral?.


In [6]:
lanzamientos = 10
opciones = 2

resultados = opciones**lanzamientos
print('Los resultados posibles son:',resultados)


1024

2. En un aeropuerto se tiene a 10 pasajeros esperando en la sala de preembarque, la polícia debe controlar a 3 de ellos. ¿Cuantas combinaciones posibles se pueden obtener?.

In [8]:
pasajeros = 10
control = 3

#Combinacion porque ABC es igual que CBA
com = math.factorial(pasajeros) / (math.factorial(control) * math.factorial(pasajeros-control))
print('Las combinaciones posibles son:',com)

Las combinaciones posibles son: 120.0


3. La Agencia Nacional de Seguridad Vial realizó una investigación para saber si los conductores de están usando sus cinturones de seguridad. Los datos muestrales fueron los siguientes:<br>
Conductores que emplean el cinturón <br>

In [79]:
data = np.array([[148,52],[162,54],[296,74],[252,48]])
df = pd.DataFrame(data,columns=['Si','No'],index=['Norte','Noreste','Sur','Centro'])
df

Si  No
Norte    148  52
Noreste  162  54
Sur      296  74
Centro   252  48

- a) ¿Qué metodo cree que se utilizo para asignar probabilidades?. <br> &emsp; &emsp; Se utilizo la corriente frecuentista <br> <br>
- b) Construya un cuadro similar, pero con la asignación de probabilidades. <br>

In [91]:
df_prob=df.copy()
#Cuadro solo con probabilidades
tot=df_prob['Si'].sum()+df_prob['No'].sum()
df_prob['Si'] = df_prob['Si']/tot
df_prob['No'] = df_prob['No']/tot
df_prob.loc[5]=[df_prob['Si'].sum(),df_prob['No'].sum()]
df_prob.rename(index={5:'Total'},inplace=True)

df_prob['Total']=df_prob['Si']+df_prob['No']
df_prob

Si        No     Total
Norte    0.136280  0.047882  0.184162
Noreste  0.149171  0.049724  0.198895
Sur      0.272560  0.068140  0.340700
Centro   0.232044  0.044199  0.276243
Total    0.790055  0.209945  1.000000

- c) ¿Cuál sería el mejor método pára estimar la probabilidad de que en Estados Unidos un conductor lleve puesto el cinturón?. <br> <br>
- d) Un año antes, la probabilidad en Argentina de que un conductor llevara puesto el cinturón era 0.75. El director de ANSV, se esperaba que la probabilidad llegara a 0.78. ¿Estará satisfecho con los resultados del estudio? <br> &emsp; &emsp; Si. La probabilidad ahora es 0.79, lo cual es mayor que 0.78 <br><br>
- e) ¿Cuál es la probabilidad de que se use el cinturón en las distintas regiones del país? ¿En qué región se usa más el cinturón?(Utilizar misma tabla que el ejercicio anterior). <br> &emsp; &emsp; La region donde se usa mas el cinturon es el Sur con un 34.07%

4. Crear una funcion que permita calcular a probabilidad de los siguientes eventos en un baraja de 52 cartas.<br>
- Obtener una carta roja.<br>
- Obtener una carta negra.<br>
- Obtener una pica.<br>
- Obtener un trébol.<br>
- Obtener un corazón.<br>
- Obtener un diamante.<br>

In [20]:
def prob_carta(tipo):
    baraja = 52
    cartas_por_color = baraja / 2
    cartas_por_palo = baraja / 4

    if tipo == 'Roja' or tipo == 'Negra':
        prob = cartas_por_color / baraja
        print('Probabilidad de obtener una carta',tipo,'es:',prob)

    elif tipo == 'Pica' or tipo == 'Trebol' or tipo == 'Corazon' or tipo == 'Diamante':
        prob = cartas_por_palo / baraja
        print('Probabilidad de obtener un',tipo,'es:',prob)
    else:
        print('El tipo de carta es invalido') 


In [21]:
prob_carta('Roja')
prob_carta('Negra')
prob_carta('Pica')
prob_carta('Trebol')
prob_carta('Corazon')
prob_carta('Diamante')

Probabilidad de obtener una carta Roja es: 0.5
Probabilidad de obtener una carta Negra es: 0.5
Probabilidad de obtener un Pica es: 0.25
Probabilidad de obtener un Trebol es: 0.25
Probabilidad de obtener un Corazon es: 0.25
Probabilidad de obtener un Diamante es: 0.25


5. La probabilidad de que salga un 7 o un 8 al seleccionar una carta de una baraja de las 52 cartas que contiene el mazo. <br>

In [93]:
baraja = 52
prob_num = 4 / baraja

prob_7_8=prob_num*2
print('La probabilidad de sacar un 7 o un 8 es:',prob_7_8)

La probabilidad de sacar un 7 o un 8 es: 0.15384615384615385


6. La probabilidad de tu país gane el mundial de fútbol.<br>

In [95]:
#Segun corriente clasica
tot_equipos=32
prob=1/tot_equipos
print('La probabilidad de que Colombia gane un mundial es:',prob)

La probabilidad de que Colombia gane un mundial es: 0.03125


7. Un experimento que tiene tres resultados es repetido 50 veces y se ve que E1 aparece 20 veces, E2 13 veces y E3 17 veces. Asigne probabilidades a los resultados.<br>

In [100]:
df_7=pd.DataFrame({'Veces':[20,13,17]},index=['E1','E2','E3'])
df_7['Probabilidad'] = df_7['Veces'] / df_7['Veces'].sum()
df_7

Veces  Probabilidad
E1     20          0.40
E2     13          0.26
E3     17          0.34